pkg/services/sqlstore/active_node.go

```diff
@@ -218,3 +229,63 @@ func GetNodeProcessingMissingAlerts(cmd *m.GetNodeProcessingMissingAlertsCommand
 	cmd.Result = &retNode
 	return nil
 }
+
+func ClusteringCleanupCheck(cmd *m.ClusteringCleanupCheckCommand) error {
+	sqlog.Debug("ClusteringCleanupCheck called")
+	lasthb := cmd.LastHeartbeat
+	results, err1 := x.Query(lastCleanupCheckSQL, lasthb-int64(setting.ClusteringCleanupPeriod), lasthb)
```

Comment:
Duration between 'lasthb-int64(setting.ClusteringCleanupPeriod)' to 'lasthb' is retention period. 
You need to check for records which are >= than 'lasthb-2*(int64(setting.ClusteringCleanupPeriod))'  and < 'lasthb-int64(setting.ClusteringCleanupPeriod) '